### ISSUE TITLE
Header level rendering changed in version 0.15

### ISSUE THREAD

**Issue by pejaham** [planet42/Laika#154]
Starting with 0.15 header rendering changed.
```
### Header 3
```
Now displays the header in `h1` size.
```
# Header 1
### Header 3
```
Displays headers as I would expect. It seems like the header size hierarchy only works when a `h1` is also used. I reverted to 0.14 which works as expected (like all other Markdown renderers I know). Is this a bug or do I overlook something?

**Comment by jenshalm** [comment]
Hello and welcome.

It's true that there were changes in 0.15, but the actual root change causing this is how titles are determined.

In previous versions, a header was only used as the title of the document when it was the **only** h1 header in the document, followed by h2 or lower headers. This was somewhat counter-intuitive and over-restrictive and not how most other site builders work. Starting with 0.15 the first header of the document is always used as the title, no matter what level it has.

The title is a special AST element called `Title`, which in contrast to the `Header` AST element does not keep level info. Instead it renders with a `title` class which you could use CSS for if you want to display it smaller: `h1.title { font-size: ... }`.

In short: the behaviour has only changed for the first header of the document, everything else is still exactly as in 0.14. And I would be somewhat reluctant to go back to the old behaviour as rendering a `Title` node as h3 seems somewhat odd to me.

Can you describe your use case and why having h3 as the root header level is important or desirable?

**Comment by pejaham** [comment]
Hi Jens,
I am working on a Play Framework application where our users can produce/contribute content via Markdown snippets (in backoffice CMS) which is rendered (via Laika) for other users - i.e. we cannot control that they do not care about AST. Sometimes they would only use h3 as the "root" size even though it is only meant as a rendering size in HTML, but not associated with an additional CSS file (so no h1.title { font-size:...} option).
We picked Laika because it seems like the most powerful library with the best documentation and it fits perfectly into our Play Scala environment. Do you think it would be possible to provide the 0.14 title behaviour via optional switch?
Thanks and regards,
Peter

**Comment by jenshalm** [comment]
I see, for the snippet use case it makes sense. I was recently mostly focussed on the "project documentation" use case, where larger sets of documents get processed.

I can introduce a flag `laika.useFirstHeaderAsTitle = true|false` and probably make `false` the default for the simple pure parser (string to string) and `true` as the default for the IO transformer that translates entire directories of inputs where you most often also end up having tables of contents and would most likely expect this behaviour to be the default.

That would be for 0.16 then which is (hopefully) due sometime in August.

**Comment by pejaham** [comment]
That's great, thanks!

Peter
